paymentsheet/src/test/java/com/stripe/android/paymentsheet/ui/PaymentOptionTest.kt

```diff
@@ -43,4 +45,31 @@ class PaymentOptionTest {
             .onParent()
             .assertContentDescriptionEquals("Card ending in 4 2 4 2 ")
     }
+
+    @Test
+    fun `remove confirmation dialog is dismissed on confirmation`() {
+        val label = "Card ending in 4242"
+
+        composeTestRule.setContent {
+            PaymentOptionUi(
+                viewWidth = 100.dp,
+                isSelected = false,
+                editState = PaymentOptionEditState.Removable,
+                isEnabled = true,
+                iconRes = R.drawable.stripe_ic_paymentsheet_card_visa,
+                description = label,
+                labelText = label,
+                onItemSelectedListener = {},
+                onRemoveListener = {},
+                onRemoveAccessibilityDescription = "remove"
+            )
+        }
+
+        // remove icon for the payment method
+        composeTestRule.onNodeWithContentDescription("remove").performClick()
+        // remove button in dialog
+        composeTestRule.onNodeWithText("Remove").performClick()
```

Comment:
Up to you - I think it could be useful to use a test tag here instead of relying on the button string